include/swift/Parse/Confusables.h

```diff
@@ -0,0 +1,23 @@
+//===--- Confusables.h - Swift Confusable Character Diagnostics -*- C++ -*-===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2015 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See http://swift.org/LICENSE.txt for license information
+// See http://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+
+#ifndef SWIFT_CONFUSABLES_H
+#define SWIFT_CONFUSABLES_H
+
+#include <stdint.h>
+
+namespace swift {
+namespace confusable {
+  uint32_t characterIsConfusable(uint32_t codepoint);
```

Comment:
Using `Is` in the name suggests that the function returns `bool`, which it does not.  I would suggest something like `tryConvertConfusableCharacterToASCII`.

Also, a doc comment wouldn't hurt (specifically documenting what the return value is if the character is not confusable).